forms/models/elements/location.js

```diff
@@ -50,7 +50,7 @@ define(function (require) {
           this.validationError = errors;
           this.trigger('invalid', this);
           return Promise.reject(error);
-        });
+        }.bind(this));
```

Comment:
with the removal of the bind, i think 
```
this.validationError = errors;
this.trigger('invalid', this);
```

should swap out `this` for `self` ??